### PR TITLE
Upgrade codemirror to 0.20

### PR DESCRIPTION
**Pull request by dralletje**
It is a sunny day here, and you know what that means:
🤘 TIME TO PROGRAM 🎸

There was a whole new codemirror version with breaking changes and al released, and I figured maybe this would help with the mixed parser issues we had.

It does use a custom version of the `@codemirror/language` module with this PR: https://github.com/codemirror/language/pull/2

Also added `PLUTO_ENABLE_CM_MIXED_PARSER()` to the javascript sneaky console features. This one will be saved to localstorage, so people who don't fear an mixed-parser error can try it out (I haven't had an error yet)

### CODE REVIEW

**Comment by dralletje** [comment]
So I found the mixed parser generates a bunch of flickering when using multi-nested languages (e.g. javascript inside html <script>s). Seems like it doesn't incrementally parse those. When using `@javascript """...` on its own, it is snappy! (even with 6x slowdown)

**Comment by fonsp** [comment]
I am getting a huge slowdown of Firefox, in the JavaScript sample notebook 😥

image

After 60 seconds, the page becomes responsive again, with this error message:

Schermafbeelding 2022-05-18 om 16 01 15

**Comment by dralletje** [comment]
@fonsp oof! Nice find, will investigate 🕵️‍♂️

**Comment by fonsp** [comment]
FYI i did some struggling to get https://github.com/JuliaPluto/codemirror-pluto-setup to generate the `.d.ts` file. Found the problem to be https://github.com/lezer-parser/highlight/pull/1, worked around that with https://github.com/JuliaPluto/codemirror-pluto-setup/commit/3688d641c173f531fe7aedc0619621d1bd7846a0 , and it now works!

**Comment by dralletje** [comment]
> FYI i did some struggling to get https://github.com/JuliaPluto/codemirror-pluto-setup to generate the `.d.ts` file. Found the problem to be [lezer-parser/highlight#1](https://github.com/lezer-parser/highlight/pull/1), worked around that with [JuliaPluto/codemirror-pluto-setup@3688d64](https://github.com/JuliaPluto/codemirror-pluto-setup/commit/3688d641c173f531fe7aedc0619621d1bd7846a0) , and it now works!

Legend

File: frontend/components/CellInput.js [hunk 1]
```diff
@@ -59,11 +61,26 @@ import { pluto_paste_plugin } from "./CellInput/pluto_paste_plugin.js"
...
+    console.log(`YOU ENABLED THE CODEMIRROR MIXED LANGUAGE PARSER
+Thanks! Awesome!
+Please let us know if you find any bugs...
+If enough people do this, we can make it the default parser.
+`)
+}
+
+// Added this so we can have people test the mixed parser, because I LIKE IT SO MUCH - DRAL
+// @ts-ignore
+window.PLUTO_ENABLE_CM_MIXED_PARSER = () => {
```
Comment (pankgeorg) [inline frontend/components/CellInput.js:80]:
```suggestion
window.PLUTO_TOGGLE_CM_MIXED_PARSER = () => {
```

Comment (dralletje) [reply frontend/components/CellInput.js:80]: Very good point hahahaha

File: frontend/components/CellInput/mixedParsers.js [hunk 2]
```diff
@@ -66,13 +66,12 @@ const overlayHack = (overlay, input) => {
 
         const result = []
         let last_content_start = from
-        for (let relative_escape_start of text.matchAll(/\\\$/g)) {
+        for (let { index: relative_escape_start } of text.matchAll(/\\\$/g)) {
```
Comment (pankgeorg) [inline frontend/components/CellInput/mixedParsers.js:69]: nice!

**Comment by fonsp** [comment]
Thanks michiel!!

**Pull request merged**